src/transport/SessionManager.cpp

```diff
@@ -79,7 +79,8 @@ SessionManager::~SessionManager()
 
 CHIP_ERROR SessionManager::Init(System::Layer * systemLayer, TransportMgrBase * transportMgr,
                                 Transport::MessageCounterManagerInterface * messageCounterManager,
-                                chip::PersistentStorageDelegate * storageDelegate, FabricTable * fabricTable)
+                                chip::PersistentStorageDelegate * storageDelegate, FabricTable * fabricTable,
+                                Crypto::SessionKeystore & sessionKeystore)
```

Comment:
Since the FabricTable is also needed by everything else that needs access to a SessionKeystore, I am curious why we do not simply make SessionKeystore belong to fabric table? Are there exceptions? Would this be very unclean? A getter on FabricTable may be less annoying that much more dependency injection on more classes

Reply:
Would this actually simplify anything? `SessionManager` seems to be initialized as many times as `FabricTable`, and PASE and CASE sessions have equal access to both objects. Also, `SessionManager` sounds like a more natural place to me since it's meant to manage per-session data while `FabricTable` - per-fabric data.